### DISCUSSION TITLE
Does this tool can scan buildSrc folder?

### DISCUSSION THREAD

**Discussion by piyush-kukadiya** [detekt/detekt#3757] [Q&A] [answered]
I have a `buildsrc` folder to manage dependencies and it's versions to reuse across modules. Issue is Android Lint tool provided by IntellIJ is unable to scan buildSrc for lint check `nerwer version of this dependency is available`
So I tried to write custom lint rules as mentioned here http://tools.android.com/tips/lint-custom-rules

But issue is custom rule will scan only on `build.gradle` or `build.gradle.kts` and all sources except `buildSrc` 

Then as an alternative I jumped to `detekt` So my question here is if I write custom rule using this tool then will it scan `buildSrc` ?

So `nerwer version of this dependency is available` can be checked by scanning `.kt` files in `buildSrc`

I hope you get my question. Thank you

**Comment by BraisGabin** [comment]
You can use detekt, sure. But I think that you will need to use the cli.

**Reply by piyush-kukadiya** [reply]
thanks👍

**Comment by cortinico** [comment, marked as answer]
You can actually also use the Detekt Gradle Plugin for your `buildSrc`.

Just go inside your build.gradle.kts file inside `buildSrc` and:

```
plugins {
    `kotlin-dsl`
    id("io.gitlab.arturbosch.detekt") version "1.17.0"
}

repositories {
    mavenCentral()
}
```

You will then be able to call `./gradlew -p buildSrc detekt`

**Reply by piyush-kukadiya** [reply]
Thanks a ton. Also I am assuming that for version check I will have to write custom rule right? or there is a built-in rule for that functionality?

**Reply by cortinico** [reply]
> Also I am assuming that for version check I will have to write custom rule right?

What do you mean with _version check_?

**Reply by piyush-kukadiya** [reply]
@cortinico I am trying to achieve [this functionality](https://android.googlesource.com/platform/tools/base/+/studio-master-dev/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/GradleDetector.kt?autodive=0%2F%2F%2F#2261) using `detekt`.

**Reply by cortinico** [reply]
Correct, you will have to write a custom rule for that. I would highly suggest you take a look at solutions like https://github.com/jmfayard/refreshVersions or similar tools.
